C7/Map/Civ3TerrainTileset.cs

```diff
@@ -0,0 +1,103 @@
+using Godot;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace C7.Map {
+
+	class TerrainPcx {
+		private static Random prng = new Random();
+		private string name;
+		// abc refers to the layout of the terrain tiles in the pcx based on
+		// the positions of each terrain at the corner of 4 tiles.
+		// - https://forums.civfanatics.com/threads/terrain-editing.622999/
+		// - https://forums.civfanatics.com/threads/editing-terrain-pcx-files.102840/
+		private string[] abc;
+		public int atlas;
+		public TerrainPcx(string name, string[] abc, int atlas) {
+			this.name = name;
+			this.abc = abc;
+			this.atlas = atlas;
+		}
+		public bool validFor(string[] corner) {
+			return corner.All(tile => abc.Contains(tile));
+		}
+		private int abcIndex(string terrain) {
+			List<int> indices = new List<int>();
+			for (int i = 0; i < abc.Count(); i++) {
+				if (abc[i] == terrain) {
+					indices.Add(i);
+				}
+			}
+			return indices[prng.Next(indices.Count)];
+		}
```

Comment:
`indices = abc.Where(i => i == terrain).Select(i => i.Index).toList()` or something like that
If selecting a random item from a collection will be a common operation, we may want an extension method for it